### PR TITLE
Add .NET Framework 4.7.2 and 4.8 to VS2017 and VS2019

### PR DESCRIPTION
**Pull request by v-chayan**
1. Include DotNet472.DeveloperTools for VS2017 installation script.
2. Add DotNet48 installation scripts.

Addresses #888

### CODE REVIEW

File: images/win/Vs2017-Server2016-Readme.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ In addition the following optional components are installed:
 * Microsoft.Net.Component.4.7.1.SDK
 * Microsoft.Net.Component.4.7.1.TargetingPack
 * Microsoft.Net.ComponentGroup.4.7.1.DeveloperTools
+* Microsoft.Net.ComponentGroup.4.7.2.DeveloperTools
```
Comment (kaylangan) [inline images/win/Vs2017-Server2016-Readme.md]: The README's are generated. You don't need to update them manually. Can you revert the changes to this file?

File: images/win/Vs2019-Server2019-Readme.md [hunk 2]
```diff
@@ -115,9 +115,9 @@ _Toolset Version:_ 3.11.2318<br/>
 _Environment:_
 * WIX: Installation root of WIX
 
-## .NET 4.7.2
+## .NET 4.8
```
Comment (kaylangan) [inline images/win/Vs2019-Server2019-Readme.md]: The README's are generated. You don't need to update them manually. Can you revert the changes to this file?

File: images/win/scripts/Installers/Vs2017/Install-VS2017.ps1 [hunk 3]
```diff
@@ -70,6 +70,7 @@ $WorkLoads = '--allWorkloads --includeRecommended ' + `
                 '--add Microsoft.Net.Component.4.7.1.SDK ' + `
                 '--add Microsoft.Net.Component.4.7.1.TargetingPack ' + `
                 '--add Microsoft.Net.ComponentGroup.4.7.1.DeveloperTools ' + `
+                '--add Microsoft.Net.ComponentGroup.4.7.2.DeveloperTools ' + `
```
Comment (kaylangan) [inline images/win/scripts/Installers/Vs2017/Install-VS2017.ps1]: Do we need to do the same in the VS2019 image as well?
https://github.com/microsoft/azure-pipelines-image-generation/blob/ca4cdf843db0b3057d9eaeff19adc0e620119f8f/images/win/scripts/Installers/Vs2019/Install-VS2019.ps1#L76

**Comment by kaylangan** [comment]
@alepauly we should probably create a related issue for testing/verifying the feature, perhaps something Sergey's team can help with. What do you think?

File: images/win/Vs2017-Server2016-Readme.md (hunk 1 above)
Comment (v-chayan) [reply images/win/Vs2017-Server2016-Readme.md]: Done.

File: images/win/Vs2019-Server2019-Readme.md (hunk 2 above)
Comment (v-chayan) [reply images/win/Vs2019-Server2019-Readme.md]: > The README's are generated. You don't need to update them manually. Can you revert the changes to this file?

Done.

File: images/win/scripts/Installers/Vs2017/Install-VS2017.ps1 (hunk 3 above)
Comment (v-chayan) [reply images/win/scripts/Installers/Vs2017/Install-VS2017.ps1]: > Do we need to do the same in the VS2019 image as well?
> 
>   
>     
>       azure-pipelines-image-generation/images/win/scripts/Installers/Vs2019/Install-VS2019.ps1
>     
>     
>          Line 76
>       in
>       ca4cdf8
>     
>     
>     
>     
> 
>         
>           
>            '--add Microsoft.Net.ComponentGroup.4.7.1.DeveloperTools ' + `

In VS2019, 4.7.2 DeveloperTools is a required component in ManagedDesktop and some other workloads. I saw that Microsoft.VisualStudio.Workload.ManagedDesktop is already included in the script so there is no need to include 4.7.2 DeveloperTools once more.

Comment (kaylangan) [reply images/win/scripts/Installers/Vs2017/Install-VS2017.ps1]: Got it.

File: images/win/vs2017-Server2016-Azure.json [hunk 4]
```diff
@@ -173,16 +173,6 @@
...
             ]
         },
-        {
-            "type": "powershell",
-            "valid_exit_codes": [
-                0,
-                3010
-            ],
-            "scripts":[
-                "{{ template_dir }}/scripts/Installers/Install-NET472.ps1"
```
Comment (kaylangan) [inline images/win/vs2017-Server2016-Azure.json]: Do we need to add a step for 4.8?

Comment (v-chayan) [reply images/win/vs2017-Server2016-Azure.json]: It is up to your design. In Jamshed's mail, he only asked me to add .NET 4.8 to VS2019. And actually, .NET 4.8 will only be inserted into VS2019 in the future. We don't have a plan to insert it to VS2017.

From: Jamshed Damkewala <Jamshed.Damkewala@microsoft.com> 
Sent: 2019年4月30日 7:28
To: Vijay Machiraju <vijay.machiraju@microsoft.com>; David Kennedy <daveken@exchange.microsoft.com>; Matt Galbraith <Matt.Galbraith@microsoft.com>; Matt Oswald <Matt.Oswald@microsoft.com>; VSTS CI Experts <VSTSCIExperts@service.microsoft.com>; Austin Jensen <Austin.Jensen@microsoft.com>; Nikola Milosavljevic (CLR) <nikolam@microsoft.com>; Chaojie Yang (Pactera Technologies Inc) <v-chayan@microsoft.com>
Subject: RE: Need to change installed .NET Desktop Frameworks on Hosted pool machines

Nikola/Chaojie, we need to update these files so the VS 2017 install includes .NET 4.7.2 targeting pack + 4.72 SDK + .NET 4.7.2 runtime, and the VS 2019 install includes .NET 4.72 targeting pack + 4.7.2 SDK and .NET 4.8 targeting pack + 4.8 SDK and .NET 4.8 runtime.

Can you please help?

Comment (kaylangan) [reply images/win/vs2017-Server2016-Azure.json]: Ok. Can we add it to the VS 2017 image too then? Thanks!

Comment (v-chayan) [reply images/win/vs2017-Server2016-Azure.json]: > Ok. Can we add it to the VS 2017 image too then? Thanks!

Done.

**Comment by mattcwebster** [comment]
@kaylangan has this been released?  You mentioned it may take time to reach my account depending on what ring I'm in.  Is there anyway to verify when I should see the changes in my account.  I just tried building a project targeting 4.8 and it failed again.

**Comment by kaylangan** [comment]
@mattcwebster no, it's not yet been released. We're currently rolling out the M151 deployment which was delayed. We plan to roll out M152 immediately after M151.

**Comment by jonatastrife** [comment]
> @kaylangan has this been released? You mentioned it may take time to reach my account depending on what ring I'm in. Is there anyway to verify when I should see the changes in my account. I just tried building a project with Azure Hosted VS2019 targeting 4.8 and it failed again.

I'm doing the same thing and still getting the errors. Just waiting for it.
